src/library-authoring/data/slice.js

```diff
@@ -0,0 +1,43 @@
+/* eslint-disable no-param-reassign */
+import { createSlice } from '@reduxjs/toolkit';
```

Comment:
Out of curiosity, did you see some advantages of using redux instead of a Context for the toaster and the sidebar?

Reply:
I have personally found the following advantages:
- We can use the values ​​and functions in any component. Unlike contex there would have to be a parent `LibraryLayout` and only children could use it.
- We directly use the reducers to update the state and we can generate specific reducers for a function (e.g. openAddContentSidebar)
- It's easier and cleaner to handle states, we don't have to do things [like this](https://github.com/openedx/frontend-app-course-authoring/blob/master/src/taxonomy/TaxonomyLayout.jsx#L16)

Reply:
> We can use the values ​​and functions in any component. Unlike context there would have to be a parent LibraryLayout and only children could use it.

Because you defined the redux state as `state.libraryHome.______`, you're already saying that it's only going to be relevant to components that are children of the library home anyways. If we have something outside of that like the Header that's accessing the internal state of the library home, it shows a problem with how the data is structured. So in this case I think the context actually forces a better separation of concerns. There's nothing wrong with having a small `<LibraryContext.Provider>` and/or `<LibraryLayout>`.

> We directly use the reducers to update the state and we can generate specific reducers for a function (e.g. openAddContentSidebar)

You can easily do that with the context version too though:
```typescript
const [currentSidebar, setCurrentSidebar] = React.useState<undefined|'add-content'>(undefined);
const context = React.useMemo(() => ({
    currentSidebar,
    openAddContentSidebar: () => setCurrentSidebar('add-content'),
    closeSidebar: () => setCurrentSidebar(),
}), [currentSidebar]);
```

> It's easier and cleaner to handle states, we don't have to do things [like this](https://github.com/openedx/frontend-app-course-authoring/blob/master/src/taxonomy/TaxonomyLayout.jsx#L16)

What's wrong with that? It is simple and has way fewer lines of code than the redux version.

Personally I much prefer the non-redux approach; it has less boilerplate and is simpler to understand.

Reply:
@bradenmacdonald Thanks for the feedback! Now I see the pros. It is true that `Context` is simpler and for our case it is fine. I will change to it :slightly_smiling_face:

Reply:
Updated here: https://github.com/open-craft/frontend-app-course-authoring/pull/38/commits/fcfa8489313f995f7f0b03caaafedc85b2ea9fec